### PR TITLE
Remove developing installation instructions from README.rst

### PR DESCRIPTION
**Pull request by dstansby**
Split out from https://github.com/sunpy/sunpy/pull/3727, where @nabobalis deleted these lines.

### CODE REVIEW

**Comment by Cadair** [comment]
Are these somewhere else? Should we link to them instead?

**Comment by nabobalis** [comment]
What do you want?

**Comment by dstansby** [comment]
Should probably link to  https://docs.sunpy.org/en/latest/dev_guide/newcomers.html#setting-up-a-development-environment ?

**Comment by nabobalis** [comment]
Works for me.

File: README.rst [hunk 1]
```diff
@@ -111,6 +94,7 @@ For more information on contributing to SunPy, please read our `Newcomers' guide
 .. _`#sunpy:openastronomy.org`: https://chat.openastronomy.org/#/room/#sunpy:openastronomy.org
 .. _issues page: https://github.com/sunpy/sunpy/issues
 .. _Newcomers' guide: https://docs.sunpy.org/en/latest/dev_guide/newcomers.html
```
Comment (nabobalis) [inline README.rst:96]:
```suggestion
```
Not sure we need this reference?

File: README.rst [hunk 2]
```diff
@@ -48,24 +48,7 @@ Developing
...
-The best way to do this is to create a new conda environment and install the git version of SunPy in it:
-
-.. warning::
-
-    Do not clone the sunpy repository into ``$HOME/sunpy``. Depending on the operating system this location is used to store downloaded data files.
-    This will cause conflicts later on, so the last argument (``sunpy-git``) on the ``git clone`` line will become the local folder name of the cloned repository.
-
-.. code:: bash
-
-    $ conda config --append channels conda-forge
```
Comment (nabobalis) [inline README.rst:60]: Why I ever wrote these I will to this day, never understand.

**Pull request merged**